executor/runtime/docker/docker_linux.go

```diff
@@ -84,6 +84,46 @@ func setupScheduler(cred ucred) error {
 	return nil
 }
 
+func isCgroup2HybridMode() bool {
+	var st unix.Statfs_t
+	err := unix.Statfs(hybridMountpoint, &st)
+	if err != nil {
+		if os.IsNotExist(err) {
+			// ignore the "not found" error
+			return false
+		}
+		panic(fmt.Sprintf("cannot statfs cgroup root: %s", err))
+	}
+	return st.Type == unix.CGROUP2_SUPER_MAGIC
+}
+
+// move container's pid 1 to unified cgroup hierarchy to enable eBPF hooks
+func movePid1ToUnifiedController(cred ucred) error {
+	if !isCgroup2HybridMode() {
+		return nil
+	}
+	cgroupPath := filepath.Join("/proc/", strconv.Itoa(int(cred.pid)), "cgroup")
+	cgroupsInfo, err := cgroups.ParseCgroupFile(cgroupPath)
+	if err != nil {
+		logrus.WithError(err).Error("Could not read container cgroups")
+		return err
+	}
+	if controllerPath, ok := cgroupsInfo["name=systemd"]; ok {
```

Comment:
Can you add a comment as to why we only do this if name=systemd?

Reply:
Wait, why do we extract out the controller path from there, and not just move it into the cgroup2 that the executor is running within?

Reply:
Likely because the executor is running under 
`1:name=systemd:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service`

and the container is under 
`name=systemd:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2/init.scope`

Reply:
So basically using `/prod/$pid/cgroup` to get the path under cgroupv1 hierarchy and recreating it under unified hierarchy.
`(nfsuper) ~ $ cat /proc/1703461/cgroup
11:hugetlb:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2
10:perf_event:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2
9:devices:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2
8:pids:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2
7:freezer:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2
6:memory:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2
5:cpuset:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2
4:cpu,cpuacct:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2
3:net_cls:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2
2:blkio:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2
1:name=systemd:/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2/init.scope
0::/containers.slice/titus-executor@default__3059b7ee-63ac-43ff-adf6-c0cf7d399a85.service/8f5534ff716e9492999b5a859cdc5342eb791e9f80d6e35eac42c81eb0c90cd2`